### PR TITLE
Implemented clients eager load

### PR DESCRIPTION
**Pull request by Ferioney**
Fixes gh-729

### CODE REVIEW

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/support/LoadBalancerEagerContextInitializer.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		this.factory = factory;
+		this.serviceNames = serviceNames;
+	}
+
+	@Override
+	public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
+		warnUpFunction();
+	}
+
+	private void warnUpFunction() {
```
Comment (spencergibb) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/support/LoadBalancerEagerContextInitializer.java:43]: no need for a separate method

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/config/LoadBalancerAutoConfiguration.java [hunk 2]
```diff
@@ -64,4 +67,10 @@ public LoadBalancerClientFactory loadBalancerClientFactory() {
 		return clientFactory;
 	}
 
+	@Bean
+	@ConditionalOnProperty(value = "spring.cloud.loadbalancer.eager-load.enabled", havingValue = "true")
```
Comment (spencergibb) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/config/LoadBalancerAutoConfiguration.java:71]: Should activation be if clients is not empty? Why set two properties?

**Review by spencergibb — CHANGES_REQUESTED** [review]

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/support/LoadBalancerEagerContextInitializer.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		this.serviceNames.forEach(factory::getInstance);
```
Comment (OlgaMaciaszek) [inline spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/support/LoadBalancerEagerContextInitializer.java:44]: Please remove unnecessary `this.` keyword.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
@Ferioney thanks for submitting the PR. Other than some minor changes described in the comments, it looks good. Please also add a documentation subsection in `spring-cloud-commons.adoc` within the "Spring Cloud LoadBalancer" section.

**Comment by OlgaMaciaszek** [comment]
@Ferioney Please also resolve conflicts.

**Comment by dplesa** [comment]
@OlgaMaciaszek or @Ferioney, can you please finish the work on this issue and merge it, since this feature is really important.

**Comment by OlgaMaciaszek** [comment]
Merged within d84c772b7e6ba9b5841449aad08384973c591ce5. Thanks again for submitting it @Ferioney .

**Pull request closed without merging**